src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -0,0 +1,137 @@
+            var context = new BinaryAnalyzerContext() { TargetUri = new Uri(@"c:\file.dll") };
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, out CompilerDataLogger logger);
+
+            string assemblies = "Microsoft.DiaSymReader (1.3.0);Newtonsoft.Json (13.0.1)";
+
+            CompilerDataLogger.s_chunkSize = 10;
+            int chunksize = CompilerDataLogger.s_chunkSize;
+            int chunkNumber = (assemblies.Length + chunksize - 1) / chunksize;
+
+            logger.Write(context, new CompilerData { CompilerName = ".NET Compiler", AssemblyReferences = assemblies });
+            sendItems.Count.Should().Be(chunkNumber + 1);
+        }
+
+        [Fact]
+        public void CompilerDataLogger_Write_ShouldNotSend_IfNoAssemblyReferences()
+        {
+            var context = new BinaryAnalyzerContext() { TargetUri = new Uri(@"c:\file.dll") };
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, out CompilerDataLogger logger);
+
+            logger.Write(context, new CompilerData { CompilerName = ".NET Compiler", AssemblyReferences = null });
+            sendItems.Count.Should().Be(1);
+        }
+
+        [Fact]
+        public void CompilerDataLogger_Constructor_ShouldThrowException_WhenForceIsDisabledAndCsvIsEnabled()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(f => f.FileExists(It.IsAny<string>())).Returns(true);
+
+            var context = new BinaryAnalyzerContext() { TargetUri = new Uri(@"c:\file.dll") };
+            var compilerOptions = new PropertiesDictionary
+            {
+                { "CsvOutputPath", @"C:\temp\" }
+            };
+
+            context.Policy = new PropertiesDictionary
+            {
+                { "CompilerTelemetry.Options", compilerOptions }
+            };
+
+            Assert.Throws<InvalidOperationException>(() => new CompilerDataLogger(SarifPath, context, fileSystem.Object));
+        }
+
+        [Fact]
+        public void CompilerDataLogger_Constructor_ShouldNotThrowException_WhenForceAndCsvAreEnabled()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(f => f.FileExists(It.IsAny<string>())).Returns(true);
+
+            var context = new BinaryAnalyzerContext() { TargetUri = new Uri(@"c:\file.dll"), ForceOverwrite = true };
+            var compilerOptions = new PropertiesDictionary
+            {
+                { "CsvOutputPath", @$"C:\temp\{Guid.NewGuid()}.sarif" }
+            };
+
+            context.Policy = new PropertiesDictionary
+            {
+                { "CompilerTelemetry.Options", compilerOptions }
+            };
+
+            var compilerDataLogger = new CompilerDataLogger(SarifPath, context, fileSystem.Object);
+            compilerDataLogger.writer.Should().NotBeNull();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_Constructor_ShouldThrowException_WhenTelemetryIsEnabledAndSarifDoesNotExist()
```

Comment:
>CompilerDataLogger_Constructor_ShouldThrowException_WhenTelemetryIsEnabledAndSarifDoesNotExist [](http://example.com/codeflow?start=20&length=94)

Can you open a bug so that we don't have to read the summary from disk?